src/interfaces/ffi/hybrid_cc_aes.rs

```diff
@@ -730,38 +677,45 @@ pub unsafe extern "C" fn h_aes_encrypt_block(
         std::slice::from_raw_parts(symmetric_key_ptr as *const u8, symmetric_key_len as usize)
             .to_vec();
 
-    // UID
-    let uid = if !uid_ptr.is_null() && uid_len > 0 {
-        std::slice::from_raw_parts(uid_ptr as *const u8, uid_len as usize).to_vec()
+    //
+    // Associated Data
+    let associated_data = if !associated_data_ptr.is_null() && associated_data_len > 0 {
+        std::slice::from_raw_parts(
+            associated_data_ptr as *const u8,
+            associated_data_len as usize,
+        )
+        .to_vec()
     } else {
         vec![]
     };
 
-    // Data
-    ffi_not_null!(data_ptr, "Data pointer should not be null");
-    if data_len == 0 {
-        ffi_bail!("The data should not be empty");
+    let ad = if associated_data.is_empty() {
```

Comment:
Here `ad` is used for `associated_data` but before it's `additional_data`. Is there a difference? (not a big fan of the abbreviation anyway…)